### PR TITLE
Bugfix/berte 530

### PR DESCRIPTION
**Pull request by bertranddemiddelaer**

### CODE REVIEW

File: bert_e/server/status.py [hunk 1]
```diff
@@ -48,16 +48,44 @@ def display():
...
-            line = {'pr_id': pr_id}
+            line = {'pr_id': pr_id, 'hotfix': False}
             for version, sha1 in queued_commits:
                 status = BUILD_STATUS_CACHE[build_key].get(sha1, None)
                 state = status.state if status else 'NOTSTARTED'
                 line[version] = {
                     'sha1': sha1,
                     'status': state,
                 }
+            if len(queued_commits) == 1 and version.count(".") == 3:
```
Comment (tcarmet) [inline bert_e/server/status.py:59]: pr can have multiple commits but I feel like it doesn't mean what I think it means

File: bert_e/workflow/gitwaterflow/branches.py [hunk 2]
```diff
@@ -583,19 +584,18 @@ def _extract_pr_ids(self, queues):
...
 
         if greatest_dev:
             for qint in queues[greatest_dev][QueueIntegrationBranch]:
-                if qint.pr_id not in prs:
+                if qint.pr_id not in prs_hf + prs:
                     prs.insert(0, qint.pr_id)
 
-        prs.sort()
-        return prs
+        return prs_hf + prs
```
Comment (tcarmet) [inline bert_e/workflow/gitwaterflow/branches.py:598]: how does this works?

File: bert_e/workflow/gitwaterflow/branches.py [hunk 3]
```diff
@@ -666,17 +666,36 @@ def queued_prs(self):
...
+                    if pr_hf_id not in pr_hf_ids:
+                        pr_hf_ids = [pr_hf_id] + pr_hf_ids
+
+        # Remove hotfix PRs from the first set
+        pr_non_hf_ids = []
+        for pr_id in pr_ids:
+            if pr_id not in pr_hf_ids:
+                pr_non_hf_ids = pr_non_hf_ids + [pr_id]
+
+        return pr_hf_ids + pr_non_hf_ids
```
Comment (tcarmet) [inline bert_e/workflow/gitwaterflow/branches.py:698]: how this one works?

**Review by tcarmet — APPROVED** [review]
:rocket:

File: bert_e/server/status.py (hunk 1 above)
Comment (tcarmet) [reply bert_e/server/status.py:59]: ah forgot about this comment in our call @bertranddemiddelaer

File: bert_e/workflow/gitwaterflow/branches.py (hunk 2 above)
Comment (tcarmet) [reply bert_e/workflow/gitwaterflow/branches.py:598]: this was explained offline, we're listing hotfix prs first and then the normal PR

File: bert_e/workflow/gitwaterflow/branches.py (hunk 3 above)
Comment (tcarmet) [reply bert_e/workflow/gitwaterflow/branches.py:698]: this was explained offline, we're listing hotfix prs first and then the normal PR

**Review by flavien-scality — APPROVED** [review]

File: bert_e/server/status.py (hunk 1 above)
Comment (bertranddemiddelaer) [reply bert_e/server/status.py:59]: this double condition will isolate a PR targeting a hotfix branch

**Review by cmeron-scality — COMMENTED** [review]
LGTM

**Pull request merged**